app/forks.go

```diff
@@ -1,15 +1,24 @@
 package app
 
 import (
+	"strings"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // BeginBlockForks is intended to be ran in.
 func BeginBlockForks(ctx sdk.Context, app *OsmosisApp) {
 	for _, fork := range Forks {
-		if ctx.BlockHeight() == fork.UpgradeHeight {
-			fork.BeginForkLogic(ctx, &app.AppKeepers)
-			return
+		if !strings.Contains(ctx.ChainID(), "e2e") {
+			if ctx.BlockHeight() == fork.UpgradeHeight {
+				fork.BeginForkLogic(ctx, &app.AppKeepers)
+				return
+			}
+		} else {
+			if ctx.BlockHeight() == TestFork.UpgradeHeight {
+				TestFork.BeginForkLogic(ctx, &app.AppKeepers)
+				return
+			}
```

Comment:
Hrmm, I want to spend a bit more time tmrw on the space of options here, this feels unappealing. (Complicating state machine to accomodate testing hooks. Feels like theres cleaner ways to achieve what we want)

Reply:
Would love advice on how to make this cleaner! Wanted something that just worked for now so we have at least some kind of test coverage since right now our CI is still testing V8

Reply:
I agree here with @ValarDragon -- let's discuss alternatives. Maybe we can use build tags or something.